Makefile

```diff
@@ -1,77 +1,96 @@
-# -*-🦕-*- Make Minishell -*-🦕-*- 
+# -*-🥥-*- Make Minishell -*-🥥-*- 
+#							by the Cocoshells 🥥
 
 NAME	= minishell
 
 # -*- Definitions of variables -*-
 
-SRCS_DIR	= srcs
-SRCS		= $(wildcard $(SRCS_DIR)/*.c)
+SRCS_DIR		= srcs
+SRCS			:= $(shell find $(SRCS_DIR) -name *.c)
 
-OBJS_DIR	= objs
-OBJS		= $(patsubst $(SRCS_DIR)/%.c, $(OBJS_DIR)/%.o, $(SRCS))
+OBJS_DIR		= objs
+OBJS			:= $(SRCS:%.c=$(OBJS_DIR)/%.o)
 
-INCLUDES	= -Iincludes
+OBJS_SUB_DIR	:= $(shell find $(SRCS_DIR) -type d)
+OBJS_SUB_DIR	:= $(OBJS_SUB_DIR:%=$(OBJS_DIR)/%)
 
-CC			= gcc
-RM			= rm -f
-MAKE		= make
-CFLAGS 		= -Wall -Wextra -Werror -fsanitize=address -g
+LIBFT_DIR		= 42_00_libft
+
+INCLUDES		= -Iincludes
+LIBRARIES		= -L${LIBFT_DIR} -lft -lreadline
+
+CC				= gcc
+RM				= rm -f
+MAKE			= make
+CFLAGS 			= -Wall -Wextra -Werror
 
 # Colors
 
-GRAY		= \e[1;30m
-GREEN 		= \e[1;32m
-DARK_GREEN	= \e[0;32m
-YELLOW		= \e[1;33m
-BLUE		= \e[1;34m
-PURPLE		= \e[1;35m
-CYAN		= \e[1;36m
-WHITE		= \e[1;37m
-NORMAL		= \e[0;37m
-END			= \e[0m
+GRAY			= \e[1;30m
+GREEN 			= \e[1;32m
+DARK_GREEN		= \e[0;32m
+YELLOW			= \e[1;33m
+BLUE			= \e[1;34m
+PURPLE			= \e[1;35m
+CYAN			= \e[1;36m
+WHITE			= \e[1;37m
+NORMAL			= \e[0;37m
+END				= \e[0m
 
 # -*- The Rules -*-
 
 #	Implicit rules
 
 # Create and compile objects files in a dedicated folder
-${OBJS_DIR}/%.o: ${SRCS_DIR}/%.c 
-			@mkdir -p objs
+$(OBJS_DIR)/%.o: %.c
+			@mkdir -p ${OBJS_DIR}
+			@mkdir -p $(OBJS_SUB_DIR)
 			@${CC} ${CFLAGS} ${INCLUDES} -c $< -o $@
 
 #	Active rules
 
-all:		${NAME} test
+all:			${NAME} instructions
 
 # Make libft + compile minishell
-${NAME}:	${OBJS}
+${NAME}:		${OBJS}
 			@printf "\n"
-			@$(CC) ${CFLAGS} ${INCLUDES} ${OBJS} -o $(NAME)
+			@${MAKE} bonus -C ${LIBFT_DIR}
+			@$(CC) ${CFLAGS} ${INCLUDES} ${LIBRARIES} ${OBJS} -o $(NAME)
 			@printf "	${WHITE}[${GREEN} Success. Compiled minishell.${WHITE}]\
 			${END} \n\n"
 
-test:		${NAME}
+instructions:	${NAME}
 			@printf "\n	📚 [${PURPLE}INSTRUCTIONS${END}] 📚\n"
-			@printf "\n🦕🍭	${YELLOW}To use minishell${END}\n"
-			@printf "	./minishell ??"
+			@printf "\n🥥🌴	${YELLOW}To use minishell${END}\n"
+			@printf "	./minishell cmd file"
 			@printf "\n\n"
 
-bonus:		${NAME} 
+test:			${NAME}
+			@bash tester/cocoshell_tester.sh		
+
+bonus:			${NAME} 
 
 #	Cleaning rules
 
 clean:
+			@${MAKE} clean -C ${LIBFT_DIR}
 			@${RM} ${OBJS} ${OBJS_BONUS}
 			@rm -rf objs
 			@printf "\n	${WHITE}[${BLUE} Cleaned minishell object files ${WHITE}]\
 			${END}\n"
 
-fclean:		clean
+fclean:			clean
+			@${MAKE} fclean_no_clean -C ${LIBFT_DIR}
```

Comment:
What's fclean_no_clean ?

Reply:
which now that I think about it is probably a very easy fix haha.

Reply:
Ah, I know, it's because the fclean rule in the makefile from minishell calls the clean rule from same makefile. 
So it cleans the files twice : first it calls the libft clean rule in the minishell clean rule. Then it calls the clean rule from the fclean of the libft file. 
So I got rid of the clean rule from the fclean of the libft file. 
Does that make sens or is my logic very weird and there's a better fix ? haha

Reply:
Ok, sry, I was confused cos I didn't notice it was a rule from libft's Makefile, then it makes sense.
I don't think it will make a difference honestly, cos since we're using the -f flag on rm, it won't throw any errors if you try to delete something that doesn't exist, it will just acknowldege its non-existance and move one. So it makes no difference at all to call fclean or fclean_no_clean from libft

Reply:
That's fair. It's just that I have text showing up saying "cleaned libft objs" twice or something. OCD lol.